crates/db_views/src/private_message_view.rs

```diff
@@ -137,3 +147,170 @@ impl JoinView for PrivateMessageView {
+
+    let jess_local_user_form = LocalUserInsertForm::builder()
+      .person_id(inserted_jess_person.id)
+      .password_encrypted("nada".to_string())
+      .build();
+
+    let _inserted_jess_local_user = LocalUser::create(pool, &jess_local_user_form)
+      .await
+      .unwrap();
+
+    let sara_timmy_message_form = PrivateMessageInsertForm::builder()
+      .creator_id(inserted_sara_person.id)
+      .recipient_id(inserted_timmy_person.id)
+      .content(message_content.clone())
+      .build();
+    let _inserted_sara_timmy_message_form = PrivateMessage::create(pool, &sara_timmy_message_form)
+      .await
+      .unwrap();
+
+    let sara_jess_message_form = PrivateMessageInsertForm::builder()
+      .creator_id(inserted_sara_person.id)
+      .recipient_id(inserted_jess_person.id)
+      .content(message_content.clone())
+      .build();
+    let _inserted_sara_jess_message_form = PrivateMessage::create(pool, &sara_jess_message_form)
+      .await
+      .unwrap();
+
+    let timmy_sara_message_form = PrivateMessageInsertForm::builder()
+      .creator_id(inserted_timmy_person.id)
+      .recipient_id(inserted_sara_person.id)
+      .content(message_content.clone())
+      .build();
+    let _inserted_timmy_sara_message_form = PrivateMessage::create(pool, &timmy_sara_message_form)
+      .await
+      .unwrap();
+
+    let jess_timmy_message_form = PrivateMessageInsertForm::builder()
+      .creator_id(inserted_jess_person.id)
+      .recipient_id(inserted_timmy_person.id)
+      .content(message_content.clone())
+      .build();
+    let _inserted_jess_timmy_message_form = PrivateMessage::create(pool, &jess_timmy_message_form)
+      .await
+      .unwrap();
+
+    let timmy_messages = PrivateMessageQuery {
+      unread_only: Some(false),
+      creator_id: Option::None,
+      page: Some(1),
+      limit: Some(20),
+    }
+    .list(pool, inserted_timmy_person.id)
+    .await
+    .unwrap();
+
+    assert_eq!(timmy_messages.len(), 3);
+
+    let timmy_unread_messages = PrivateMessageQuery {
+      unread_only: Some(true),
+      creator_id: Option::None,
+      page: Some(1),
+      limit: Some(20),
+    }
+    .list(pool, inserted_timmy_person.id)
+    .await
+    .unwrap();
+
+    assert_eq!(timmy_unread_messages.len(), 2);
+
+    let timmy_sara_messages = PrivateMessageQuery {
+      unread_only: Some(false),
+      creator_id: Some(inserted_sara_person.id),
+      page: Some(1),
+      limit: Some(20),
+    }
+    .list(pool, inserted_timmy_person.id)
+    .await
+    .unwrap();
+
+    assert_eq!(timmy_sara_messages.len(), 2);
+
+    let timmy_sara_unread_messages = PrivateMessageQuery {
+      unread_only: Some(true),
+      creator_id: Some(inserted_sara_person.id),
+      page: Some(1),
+      limit: Some(20),
+    }
+    .list(pool, inserted_timmy_person.id)
+    .await
+    .unwrap();
+
+    assert_eq!(timmy_sara_unread_messages.len(), 1);
```

Comment:
asserts should also check that the ids of returned messages match the ones inserted above.